### PR TITLE
DOP-2204 - {SPIKE} - [CLS] | Investigate changes in page render (mobile)

### PR DESCRIPTION
**Pull request by carooobregon**
### Stories/Links:

DOP-2204

### Staging Links:

[Docs Staging
](https://docs-mongodbcom-staging.corp.mongodb.com/master/docs/carolinao/DOP-2204/installation/)

### Notes:
The "on this page" section was taking additional time rendering on mobile because it was rendering under the `Heading` component that uses `useScreenSize()` to determine what type of device the user is currently using. Changed it so that `Content` gets rendered directly by `ComponentFactory` and determines device size based on CSS media queries. 

Additionally, refactored CSS inside `Content` component so it uses emotion Styled to set up CSS attributes in elements instead of adding them inline.

### CODE REVIEW

**Review by casthewiz — COMMENTED** [review]
I think this is good, but I'm curious if we see similar performance gains by retaining the removal of `useScreenSize()` from `Contents` and placing `Contents` back into `Heading`. 

I think we should be fine, but we are introducing an assumption in the application that `Heading` will be present in the AST for the component factory to render it - and we could de-risk this if we can keep Heading unchanged.

**Comment by carooobregon** [comment]
@casthewiz reduced time came from removing `Contents` from the `ConditionalWrapper` since the condition inside the wrapper is what is causing the delay (as it uses `useScreenSize()`), however I could place it back to Heading since the logic of how `Content` should be rendered depending on screen size is inside of its own component now if that would be better?

File: src/templates/document.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ const Document = ({
       </div>
       <RightColumn>
         <TabSelectors />
-        <Contents />
+        <Contents inRightColumn={true} />
```
[the remark this answers is not in the document]
Comment (carooobregon) [reply src/templates/document.js:72]: sounds good!

**Comment by casthewiz** [comment]
@carooobregon I think it'd be worth investigating. 
If we see the performance regress by changing it back, then we can keep `Contents` seperate from `Header` 🤔

**Comment by carooobregon** [comment]
@casthewiz moved `Contents`  to `Heading` and performance stayed the same 👍   Just pushed the new changes to the PR

**Review by casthewiz — APPROVED** [review]

**Pull request merged**